clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
@hachikuji Thanks for the reference. 

I am not aware that we have recommended user to explicitly seek beyond bad message if `consumer.poll(...)` throws exception. I didn't find it in the Java doc of `KafkaConsumer.poll(...)`. Do you know where it is documented?

I am also not sure that the current implementation of KafkaConsumer will enter infinite loop if there is a single bad record. The reason is that `PartitionRecords.nextFetchedRecord()` will always increment the iterator across the bad record before any exception is thrown due to bad record. The bad record should already be skipped even without the change in this patch. Did I miss something here?